include/qt_atomics.h

```diff
@@ -6,6 +6,19 @@
 #include <qthread/common.h>
 #include <qthread/qthread.h>
 
+#ifndef __STDC_NO_ATOMICS__
+#define USE_C11_MEMORY_FENCE
+#include <stdatomic.h>
+#endif
```

Comment:
I'm not sure how much this helps with portability. There's only a few compilers that actually set `__STDC_NO_ATOMICS__` (I think intel for a release or 2 and PGI for a while) and obviously compilers that were out before C11 was introduced won't have this set.

I don't think this is an issue for Chapel, as we now require fairly modern C compilers, but it may hurt overall qthreads portability. Maybe this could just be a configure test instead?

Reply:
We could add a configure check in favor of less ifdefs in the code. The effect would be the same. Clang and MSVC do set this variable.

Reply:
> 

To support earlier compilers, we could guard on __STDC_VERSION__ (if you don't want to work with the configure test)

Reply:
Have we decided if we are going to guard with `__STDC_VERSION__` or `__STDC_NO_ATOMICS`

Reply:
Here's the check we have in our C11 atomic implementation:

```c
#if __STDC_VERSION__ >= 201112L && !defined(__STDC_NO_ATOMICS__)
  #include <stdatomic.h>
  // ...
#endif
```

Reply:
Right, the method quoted by @ronawho is much preferred, since it filters out implementations that are not required to supply `__STDC_NO_ATOMICS__`.

Reply:
Nice. I am adding this to the PR.

Reply:
Done.